src/services/mars-photo-service.js

```diff
@@ -0,0 +1,46 @@
+const axios = require('axios');
+const { format } = require('date-fns');
+const Exception = require('../common/Exception');
+const logger = require('../common/logger');
+const config = require('../config/config');
+
+const { apiKey, baseUrl, marsPhotoEndpoint, marsManifestEndpoint } = config.nasaApi;
+const DATE_FORMAT = 'yyyy-MM-dd';
+
+const handlePhotoResponse = (response) => {
+    if (!response.data) {
+        throw new Exception(400, 'No data in NASA response');
```

Comment:
400 error covers request mainly, not upstream server behavior. So 4xx error can be returned from validator. In this case it would be better to a) retry with another date (according to task - we should find most recent photo) and b) check upstream server error code and return it